### PR TITLE
docs: Mike tabular data index revisions

### PR DESCRIPTION
**Pull request by Shoeboxam**

### CODE REVIEW

**Comment by Shoeboxam** [comment]
* **#1864** <a href="https://app.graphite.dev/github/pr/opendp/opendp/1864?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Shoeboxam and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/source/getting-started/tabular-data/index.rst [hunk 1]
```diff
@@ -86,18 +85,30 @@ The compositor is the foundation of our differentially private queries. It essen
            (): dp.Margin(max_partition_length=60_000_000),
        },
    )
+   # Once you construct the context, you should abstain from directly accessing your data again.
+   # In fact, it is good practice to delete it! 
+   del df
```
Comment (mccalluc) [inline docs/source/getting-started/tabular-data/index.rst:90]: (github won't let me suggest the whole block; sorry.)
```suggestion
.. init code that we don't want cluttering the example:
    >>> import polars as pl
    >>> df = pl.LazyFrame()

.. code-block:: python

    >>> context = dp.Context.compositor(
    ...     data=df,
    ...     privacy_unit=dp.unit_of(contributions=36),
    ...     privacy_loss=dp.loss_of(epsilon=1.0),
    ...     split_evenly_over=10,
    ...     margins={
    ...         ("YEAR", ): dp.Margin(max_partition_length=60_000_000, max_partition_contributions=4),
    ...         ("YEAR", "QUARTER",): dp.Margin(max_partition_length=60_000_000, max_partition_contributions=1),
    ...         (): dp.Margin(max_partition_length=60_000_000),
    ...     },
    ... )
    
    >>> # Once you construct the context, you should abstain from directly accessing your data again.
    >>> # In fact, it is good practice to delete it! 
    >>> del df
```

Also: In the notebook conversions, I've been moving code into tabs, even when we only have Python right now. For consistent style, would we like to say that it should also be wrapped in a tab? Or maybe that's gratuitious?

**Review by mccalluc — COMMENTED** [review]
and make the code a doctest so it will be executed.

File: docs/source/getting-started/tabular-data/index.rst (hunk 1 above)
Comment (Shoeboxam) [reply docs/source/getting-started/tabular-data/index.rst:90]: I'm fine with tabs, I expect we'll be adding more as we go.

**Review by mccalluc — COMMENTED** [review]
Screenshot 2024-08-01 at 12 00 48 PM

Unless it's intentional, can you only indent the `max_partition_length` section by two spaces, instead of the four? Screen shot on the left is the current state with four spaces, and an extra box, and the right is with two spaces.

File: python/src/opendp/extras/polars/__init__.py [hunk 2]
```diff
@@ -644,14 +644,6 @@ def agg(
...
             lf_plan = self._lgb_plan.agg(*aggs, **named_aggs)
             return LazyFrameQuery(lf_plan, self._query)
-        
-
-        def len(self, name: str | None = None) -> LazyFrameQuery:
-            """
-            Return the number of rows in each group.
-            """
-            lf_plan = self._lgb_plan.len(name=name)
-            return LazyFrameQuery(lf_plan, self._query)
```
Comment (Shoeboxam) [inline python/src/opendp/extras/polars/__init__.py:654]: This is triggering coverage issues for some reason. Not important to keep right now.

**Comment by mccalluc** [comment]
Thinking a little more: Will this require the polars extra to run? Should it? There's nothing here that says `polars` to the user. I guess it's good that the tests are surfacing this.

File: docs/source/getting-started/tabular-data/index.rst [hunk 3]
```diff
@@ -52,52 +59,103 @@ The `User Guide <https://ec.europa.eu/eurostat/documents/1978984/6037342/EULFS-D
...
+-------------------
+The compositor is the foundation of our differentially private analysis. 
+It mediates access to the sensitive data,
+ensuring that queries you would like to release satisfy necessary privacy properties. 
+
+.. testsetup::
+    >>> import polars as pl
+    >>> df = pl.LazyFrame()
+
+.. doctest:: python
```
Comment (mccalluc) [inline docs/source/getting-started/tabular-data/index.rst:118]: As long as we're clear that sphinx doesn't have a role in the doc testing, great: If I'm confused, let me know.

**Review by mccalluc — DISMISSED** [review]
some comments, but render looks fine, and the explanation of the polars dependency is good.

**Comment by GKD-stack** [comment]
> Screenshot 2024-08-01 at 12 00 48 PM
> Unless it's intentional, can you only indent the `max_partition_length` section by two spaces, instead of the four? Screen shot on the left is the current state with four spaces, and an extra box, and the right is with two spaces.

ok! it was two spaces ahead, and i made it one space

File: docs/source/getting-started/assessing-utility/noise.ipynb [hunk 4]
```diff
@@ -184,8 +184,8 @@
     "    privacy_loss=dp.loss_of(epsilon=1.0),\n",
     "    split_evenly_over=10,\n",
     "    margins={\n",
-    "        (\"ILOSTAT\", ): dp.Margin(public_info=\"keys\", max_partition_length=60_000_000),\n",
-    "        (): dp.Margin(max_partition_length=60_000_000),\n",
+    "        (\"ILOSTAT\", ): dp.polars.Margin(public_info=\"keys\", max_partition_length=60_000_000),\n",
+    "        (): dp.polars.Margin(max_partition_length=60_000_000),\n",
```
[the remark this answers is not in the document]
Comment (Shoeboxam) [reply docs/source/getting-started/assessing-utility/noise.ipynb:188]: yep

**Review by mccalluc — COMMENTED** [review]
The latest changes here broke the sphinx rendering. 
Screenshot 2024-08-01 at 6 00 21 PM

I'll push a fix to this branch, but let's be careful with any more changes to the spacing here without previewing in a local docs build.

**Review by mccalluc — APPROVED** [review]
This is sort-of a self review, but the local render looks good, so as soon as CI confirms that it is good, let's merge.

**Pull request merged**